src/Codec/Xlsx/Types.hs

```diff
@@ -28,29 +29,37 @@ import           Data.Map (Map)
 import qualified Data.Map as M
 import           Data.Text (Text)
 import qualified Data.Text as T
-
+import           Text.XML (Name)
 
 -- | Cell values include text, numbers and booleans,
 -- standard includes date format also but actually dates
 -- are represented by numbers with a date format assigned
 -- to a cell containing it
-data CellValue = CellText   Text
-               | CellDouble Double
-               | CellBool   Bool
+data CellValue = CellText    Text
+               | CellDouble  Double
+               | CellBool    Bool
                deriving (Eq, Show)
 
--- | Currently cell details include only cell values and style ids
--- (e.g. formulas from @\<f\>@ and inline strings from @\<is\>@
--- subelements are ignored)
+data CellFormula = CellFormula { _cellFormulaValue :: Maybe Text
+                               , _cellFormulaAttrs :: [(Name, Text)] 
```

Comment:
So this relates to the shared formula thing. These might be created when you "drag down" a formula in Excel. The xml of the formula of the first cell then looks something like  ```<f t="shared" ref="E2:E11" si="0">C2+D2/C3<\f>```, and the following cells look like ```<f t="shared" si="0"/>
```. We need to parse the "t", "si", and "ref" attributes. I suppose it might make more sense to create some kind of SharedFormula datatype to do so. If we don't do this, then parsing and then immediately writing a worksheet with shared formula will remove all but the first, leaving only the values.